pkg/backends/datadog/datadog.go

```diff
@@ -203,6 +206,32 @@ func (d *Client) processMetrics(metrics *gostatsd.MetricMap, cb func(*timeSeries
 		fl.maybeFlush()
 	})
 
+	metrics.Timers.Each(func(key, tagsKey string, timer gostatsd.Timer) {
+		if timer.Buckets != nil {
+			var keys []int
+			for k := range timer.Buckets {
+				keys = append(keys, k)
+			}
+			sort.Ints(keys)
+
+			previousBucket := 0
+			for _, bucket := range keys {
+				val := timer.Buckets[bucket]
+				var bucketMax string
+				if bucket == statsd.InfinityBucketSize {
+					bucketMax = "+Inf"
+				} else {
+					bucketMax = strconv.Itoa(bucket)
+				}
+				bucketTag := "between:" + strconv.Itoa(previousBucket) + "_" + bucketMax
+				newTags := timer.Tags.Concat([]string{bucketTag})
+				fl.addMetricf(counter, float64(val), timer.Hostname, newTags, "%s.buckets", key)
```

Comment:
@tiedotguy said this should be a gauge. Are we sure that counter is ok?

Reply:
We can iterate and see what works